### ISSUE TITLE
Crash on open recently played file

### ISSUE THREAD

**Issue by smaragdus** [422658476/MPV-EASY-Player#3]
**MPV-EASY Player** version **0.29.0.2**  (x64) crashes when I I try to load a file from 'open recently played file' menu. This happens under Windows 8 x64.

**Comment by 422658476** [comment]
Thank you very much for replying to this question again. This function has tested many different paths and has no problem. If you encounter this problem, I believe that it will not be encountered when opening each path.
When you encounter open problems, most of the problems are related to the path.
If you do not reveal your privacy, please provide the following information:
1. Give me the path to the crash file, such as F:\movie\xxx\xxxxx.mkv

**Comment by smaragdus** [comment]
Hello,

Please find below the full paths:

D:\Torrents\Films\Akira Kurosawa - Hakuchi (1951)\Akira Kurosawa - Hakuchi (1951).avi
D:\Films\Ana y los Lobos\Ana_y_los_lobos_(1973).avi
D:\Films\Devils on the Doorstep\Devils.on.the.Doorstep.2000.Allzine.avi
D:\Torrents\Films\Akira Kurosawa - Ikiru (1952)\Akira Kurosawa - Ikiru (1952).avi
D:\Torrents\Films\Akira Kurosawa - Dersu Uzala (1975)\Dersu Uzala (1975) cd2.avi

**MPV-EASY Player** version **0.29.0.2** (x64) crashed on all of these files.

One more thing which may be related to the same problem- when I specify 'advanced playlist save path it looks like that:

`C:\\Portable Programs\\MPV-EASY Player\\data\\playlists\\`

I think it should be:

`C:\Portable Programs\MPV-EASY Player\data\playlists`

Why there are double brackets?

**Comment by 422658476** [comment]
ok,i will test these paths tomorrow.
advanced playlist save path is ok,because lua will cover \\ to \.if only use \,save will fail.

**Comment by 422658476** [comment]
I am very sorry, the problem you said cannot be reproduced on my system.
I also think that this normal path will not cause problems, because the path complexity is much higher than this when testing.
I think this is caused by a problem with the video file itself.
If you can, please help me confirm the correctness of my judgment:
1. Transfer these files that failed to play to the root directory of the d drive and rename them to play. Check if the open still fails.

D:\Torrents\Films\Akira Kurosawa - Hakuchi (1951)\Akira Kurosawa - Hakuchi (1951).avi
 **---> D:\1.avi**

D:\Films\Ana y los Lobos\Ana_y_los_lobos_(1973).avi
 **---> D:\2.avi**

If it still fails to open, it is a problem with the video file. If it is successful, let's discuss other factors that may cause playback failure.

**Comment by smaragdus** [comment]
Hello,

Moving and renaming files and then opening them from 'open recently played file' menu worked for a while but then after several restarts the player started crashing again.

More- when it was still working opening recent files resulted in strange resolution, screens:

![mpv-easy player 0 29 0 2 - 2018-08-08 - error - 003](https://user-images.githubusercontent.com/9025757/43806314-12d63c7c-9aac-11e8-865b-5778dbfeaa77.png)

![mpv-easy player 0 29 0 2 - 2018-08-08 - error - 004](https://user-images.githubusercontent.com/9025757/43806321-1f091c9e-9aac-11e8-9140-98a8134196a5.png)

I used the following paths:

D:\1.avi
D:\2.avi

**Comment by 422658476** [comment]
Hello, this result is too different from what I expected. 
1. Since this path can be opened, it should be played normally in the previous path, sometimes it crashes after playback, instead of crashing every time.
Is this the case? Still the original path has never worked properly.
2. Will the error window pop up when the player crashes? Please give me a screenshot if you have one.
3. If you can, please provide me with one of the bt seeds of one of the files. I want to test the original file myself. Thank you very much.

**Comment by smaragdus** [comment]
Hello again,

In fact MPV-EASY Player has never crashed when I was playing video files, all video formats- AVI, FLV, MP4, etc, are playing fine.

When MPV-EASY Player crashes there is no error message.

One more thing that might give you some clue- I deleted history and dragged and dropped these two AVI files from the original location. I was able to re-open them from 'open recently played file' menu. Then I opened 3 files (FLV, MP4) from another path and using 'open recently played file' menu I was able to open the first 2 AVI files, but MPV-EASY Player crashed again when I tried to play the other 3 files from 'open recently played file' menu. So I suppose that the problem is not the video files but the paths. Also I don't think that MPV-EASY Player should load 'welcome' in 'open recently played file' menu.

I did many tests and this is what I have discovered:

- If I play videos from one directory, for example: 'D:\Films', I can load these files on next run from 'open recently played file' menu;
- If I play video files from different directories- for example 'D:\Films', 'D:\Torrents\Films', 'D:\Videos', then MPV-EASY Player crashes;

In short- I can load video files using ''open recently played file' menu if these videos are from a single root directory. If I try to load videos from different root directories, then MPV-EASY Player crashes. This is what I think. The type of the file does not matter, it seems that only paths matter.

You may try to play files from different locations (root paths) and then load them from 'open recently played file' menu.

**Comment by 422658476** [comment]
When MPV-EASY Player crashes there is no error message.
If this is the case every time, then the problem is the path.
That is to say, it is now possible to rule out the problem of being a file.

But the crash situation you said is very strange. There is no such problem in theory. What you mean is that if you drag and drop to open five files, there will be such a phenomenon:

'open recently played file' menu:
e:\xx\c.avi
e:\xx\d.avi
e:\xx\e.avi
d:\a.avi
d:\b.avi

The three files that exist under the e disk will fail to play?

In fact, as long as the file [x.avi] can be successfully opened in the [Open recently played files] menu, no matter what other files you open later, the file [x.avi] can be loaded again because Its path will not change, the path is extracted correctly, and the phenomenon you said cannot be successfully reproduced on my system.

The only way to solve this problem now is:

1. When you close the mpv-easy player, clear all history.
2. Open no more than 5 files in the same way that they will crash.
3, then package your [mpv-easy player\data\mpv-easy-data\watch_later], [watch_later] folder into a zip
4. Since you can upload your screenshot, please upload this zip package and provide it with the download address.
5. Tell me which of these 5 files will cause a crash.
6, I will check the history file of these files in the watch_later folder that should be crashed, check the problem.

**Comment by smaragdus** [comment]
I do not know what to say, perhaps my speculation is wrong- MPV-EASY Player  sometimes crashes on trying to load recent files, other times it doesn't crash.

Below I will attach watch_later archive:

[watch_later.zip](https://github.com/422658476/MPV-EASY-Player/files/2268679/watch_later.zip)

MPV-EASY Player has never crashed when adding and playing files.

Something unrelated- what about MPV-EASY Player remembering last used folder when using 'open file...' dialog?

Regards

**Comment by smaragdus** [comment]
If you need all 'data' folder I can also compress and upload it so that you will have all my settings if this is needed.

**Comment by 422658476** [comment]
I have tested it, there is no problem with the path, but I have already thought of using another way to solve this strange problem.
I will change the way [Open recently played files] works in the next version:
1. The current working mode is: close the current window and re-run mpv-easy player.exe to open the history file.
2. The way of working in the future: directly load the history file that needs to be opened without closing the current window. This should not have such a problem.
The new version will be available next Monday. If this problem occurs at the time, please feedback again.
Many thanks.

**Comment by smaragdus** [comment]
Your ideas sound good, yes, there is no need to close the player when loading a file from history.
If there are test builds you may give me a link and I will test them.
Regards

**Comment by 422658476** [comment]
[MpvEasyGui.zip](https://github.com/422658476/MPV-EASY-Player/files/2269149/MpvEasyGui.zip)

you can download MpvEasyGui.zip，Use MpvEasyGui.exe in zip to overwrite your existing file with the same name and test with mpv-easy borderless gui mode

**Comment by smaragdus** [comment]
Hello there,

I finally tested version **0.29.0.3** and I can confirm that recently played files are now opened fine. Good work, thanks.

I came upon a small problem- when i drag and drop a video with a higher resolution the player (borderless GUI) resizes in such a way that it gets out of screen- please see [examle](https://s20.postimg.cc/x7xx8jo2l/MPV-_EASY_Player_0.29.0.3_-_2018-08-18_-_error_-_001.png). So I have to manually drag player window to make it fit to screen. I think that the player should be centered to screen when it resizes according to video resolution. Please confirm this behaviour:

- open the player;
- drag&drop a high resolution video into the player;
- the player gets out of screen borders;

Any high resolution YouTube video would be fine for testing.

Regards

**Comment by 422658476** [comment]
![Hope to center.png](https://user-images.githubusercontent.com/37802190/44296010-4f840f80-a2e7-11e8-83d0-c242bcdb4113.png)

1. Compatible with mpv native gui. Keep the experience consistent. Under the mpv native gui, when you drag and drop to open a new file, it just resizes and does not automatically center.
2. Maintain consistency with the behavior expected by the user. When you drag and drop a file into the window, the user's mind is just to open a file, and there is no intention to center it. Therefore, the centering behavior is an unexpected behavior, so the automatic centering behavior is canceled.
3. Avoiding automatic centering has turned from humanization to a problem.
When you use a playlist to play many video files with different resolutions, and adjust the playback window position and size, and you want the window to be displayed in the lower right corner of the display, you definitely don't want the player to automatically jump to the middle of the display every time when you play the next file.

So for the above reasons or more, this compromise has been made, and I don't want to add a setting to improve this behavior, because this is definitely not valid for the mpv native gui.

If you feel that this behavior is not comfortable now, there is actually no solution, because the method has been given, but you may not find it.

In fact, when you drag and drop a file into the window, its size changes but the position is not centered, please click the resolution text on the control interface: [****x***], it will help you automatically center And adjust to the largest suitable size.

That is, you only need to perform window movement and mouse click after dragging and dropping.
